tar.go

```diff
@@ -135,18 +136,22 @@ func tarFile(tarWriter *tar.Writer, source, dest string) error {
 		baseDir = filepath.Base(source)
 	}
 
-	return filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+	return filepath.Walk(source, func(fpath string, info os.FileInfo, err error) error {
 		if err != nil {
-			return fmt.Errorf("error walking to %s: %v", path, err)
+			return fmt.Errorf("error walking to %s: %v", fpath, err)
 		}
 
-		header, err := tar.FileInfoHeader(info, path)
+		header, err := tar.FileInfoHeader(info, fpath)
 		if err != nil {
-			return fmt.Errorf("%s: making header: %v", path, err)
+			return fmt.Errorf("%s: making header: %v", fpath, err)
 		}
 
 		if baseDir != "" {
-			header.Name = filepath.ToSlash(filepath.Join(baseDir, strings.TrimPrefix(path, source)))
+			name, err := filepath.Rel(source, fpath)
```

Comment:
Any chance you can add a test for this here, that fails without this diff, and passes with this diff?